### DISCUSSION TITLE
Using mutateFormDataBeforeFill doesn't work on view

### DISCUSSION THREAD

**Discussion by pedrocoelho98** [filamentphp/filament#3173] [Help] [answered]
Sorry for the probable noob question, but I'm trying to show the user age on view/edit form.

What would be the best option?

I can hardcode the "age" attribute using the mutateFormDataBeforeFill but it only works on edit, not on view page...

On the View page:
`    protected function mutateFormDataBeforeFill(array $data): array
    {
        $data['age'] = '33';
        return $data;
    }`

On the form:
`TextInput::make('age')`

Also, how can we access the model field 'dob' on the resource page?

The documentation is very minimal on this.
Thanks

**Comment by danharrin** [comment, marked as answer]
```php
Placeholder::make('age')->content(fn ($record) => calculateAge($record?->dob))
```

like that? obviously replacing `calculateAge()`

**Reply by pedrocoelho98** [reply]
Thanks @danharrin.

That obviously does the trick!

And how would you do it in the Table?

**Reply by danharrin** [reply]
```
TextColumn::make('age')->getStateUsing(fn ($record) => calculateAge($record?->dob))
```

**Reply by pedrocoelho98** [reply]
Thanks a lot. I eventually found this last one on the 30th google search
